pkg/controller/transfer/server/setup.go

```diff
@@ -112,12 +112,36 @@ func postCreate(_ context.Context, cr *svcapitypes.Server, obj *svcsdk.CreateSer
 func preCreate(_ context.Context, cr *svcapitypes.Server, obj *svcsdk.CreateServerInput) error {
 	obj.Certificate = cr.Spec.ForProvider.Certificate
 	obj.LoggingRole = cr.Spec.ForProvider.LoggingRole
-	obj.EndpointDetails = &svcsdk.EndpointDetails{
-		AddressAllocationIds: cr.Spec.ForProvider.CustomEndpointDetails.AddressAllocationIDs,
-		SecurityGroupIds:     cr.Spec.ForProvider.CustomEndpointDetails.SecurityGroupIDs,
-		SubnetIds:            cr.Spec.ForProvider.CustomEndpointDetails.SubnetIDs,
-		VpcEndpointId:        cr.Spec.ForProvider.CustomEndpointDetails.VPCEndpointID,
-		VpcId:                cr.Spec.ForProvider.CustomEndpointDetails.VPCID,
+	obj.EndpointDetails = &svcsdk.EndpointDetails{}
```

Comment:
Wouldn't checking for `cr.Spec.ForProvider.CustomEndpointDetails != nil` be enough to safely assign directly?

Reply:
the problemhere for direct assign is a nil pointer - looks like that safely assign is not possible - but i'am open for other implementation if someone has an idea

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x4c7af01]

goroutine 2877 [running]:
github.com/crossplane/provider-aws/pkg/controller/transfer/server.preCreate({0xc000acd440, 0xc001e86bc8}, 0xc000acd440, 0xc001bd2700)
        /Users/haarchri/Documents/provider-aws/pkg/controller/transfer/server/setup.go:117 +0xa1
```

Reply:
I thought the `cr.Spec.ForProvider.CustomEndpointDetails` is a pointer field but it seems like it's not, so the check wouldn't make much sense. But then why do we need to change this function at all? Seems like the source and target field types are still the same on both sides.